### PR TITLE
podofo: fix library linkage on Darwin

### PR DESCRIPTION
**Pull request by mroi**
The podofo package is currently broken on Darwin: the podofo tools do not find libpodofo.0.9.6.dylib at runtime, because the library path the binary is linked against is incomplete.

The reason is that podofo builds both the library and the tools using this library in the same `cmake` run. And since the library is not yet installed when the tools are build, it does not contain the final store path as its install name. Linking the tools therefore picks up this incorrect install name.

I fixed this with a fixupPost script. I am not sure this is the correct way, since I am rather new to Nix, but the package works on Darwin now and this should have no effect for Linux.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by LnL7** [comment]
This is really strange, the dylib does have a correct install_name. The fact that the binaries built before the libraries are installed is not a problem and exactly what the install_name is for. It's used by the linker to allow linking against absolute paths while the library isn't installed in it's final location yet.

**Comment by mroi** [comment]
The dylib only receives its correct install_name when it is installed. The binaries got built before that, where the install_name of the library was still containing only the filename without any path information. At least that’s what happens when I observed the build. I am not a cmake expert, though.

I guess it works fine when you only build a library and then another project links against that library. But doing both in the same cmake build appears not to work properly.

**Comment by LnL7** [comment]
Hmm, that's unfortunate. Guess they must be using some nonstandard build scripts because I've not seen this before with cmake builds.

File: pkgs/development/libraries/podofo/default.nix [hunk 1]
```diff
@@ -30,6 +31,12 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = "-DPODOFO_BUILD_SHARED=ON -DPODOFO_BUILD_STATIC=OFF";
 
+  postFixup = if stdenv.isDarwin then ''
```
Comment (LnL7) [inline pkgs/development/libraries/podofo/default.nix]: nitpick, use `stdenv.lib.optionalString` instead of an if statement.

**Review by LnL7 — APPROVED** [review]

**Comment by mroi** [comment]
Thanks a lot for this tip. I am rather new to Nix, so I did not know this about `stdenv.lib.optionalString`. I fixed the pull request.

**Comment by LnL7** [comment]
No big deal 😄. It does about the same thing, but it's slightly nicer to use IMHO.

**Comment by mroi** [comment]
I agree. Thanks for making contributing to Nix such a nice experience.

**Pull request merged**